core/src/main/java/net/opentsdb/storage/MockDataStore.java

```diff
@@ -0,0 +1,344 @@
+  public static final long HOSTS = 4;
+  public static final long INTERVAL = 60000;
+  public static final long HOURS = 24;
+  public static final List<String> DATACENTERS = Lists.newArrayList(
+      "PHX", "LGA", "LAX", "DEN");
+  public static final List<String> METRICS = Lists.newArrayList(
+      "sys.cpu.user", "sys.if.out", "sys.if.in", "web.requests");
+
+  /** <ID, <type, list of hourly buckets>> */
+  private Map<TimeSeriesId, Map<TypeToken<?>, MockSpan<?>>> database;
+  
+  @Override
+  public Deferred<Object> initialize(final TSDB tsdb) {
+    database = Maps.newHashMap();
+    generateMockData(tsdb);
+    return Deferred.fromResult(null);
+  }
+  
+  @SuppressWarnings("unchecked")
+  @Override
+  public Deferred<Object> write(final TimeSeriesValue<?> value, 
+                                final TsdbTrace trace,
+                                final Span upstream_span) {
+    Map<TypeToken<?>, MockSpan<?>> type = database.get(value.id());
+    if (type == null) {
+      type = Maps.newHashMap();
+      database.put(value.id(), type);
+    }
+    
+    MockSpan<?> span = type.get(value.type());
+    if (span == null) {
+      if (value.type() == NumericType.TYPE) {
+        span = new MockSpan<NumericType>();
+      }
+      
+      type.put(value.type(), span);
+    }
+    
+    if (value.type() == NumericType.TYPE) {
+      ((MockSpan<NumericType>) span).addValue((TimeSeriesValue<NumericType>) value);
+    }
+    return Deferred.fromResult(null);
+  }
+
+  @Override
+  public QueryExecution<IteratorGroups> runTimeSeriesQuery(final QueryContext context,
+                                                           final TimeSeriesQuery query, 
+                                                           final Span upstream_span) {
+    class Ex extends QueryExecution<IteratorGroups> {
+
+      public Ex(TimeSeriesQuery query) {
+        super(query);
+      }
+
+      @Override
+      public void cancel() {
+        
+      }
+      
+      public void execute() {
+        setSpan(context, 
+                MockDataStore.class.getSimpleName(), 
+                upstream_span,
+                TsdbTrace.addTags(
+                    "order", Integer.toString(query.getOrder()),
+                    "query", JSON.serializeToString(query),
+                    "startThread", Thread.currentThread().getName()));
+        
+        int matches = 0;
+        final IteratorGroups results = new DefaultIteratorGroups(); 
+        for (final Entry<TimeSeriesId, Map<TypeToken<?>, MockSpan<?>>> entry : 
+              database.entrySet()) {
+          for (Metric m : query.getMetrics()) {
+            if (Bytes.memcmp(m.getMetric().getBytes(Const.UTF8_CHARSET), 
+                             entry.getKey().metrics().get(0)) != 0) {
+              continue;
+            }
+            
+            if (!Strings.isNullOrEmpty(m.getFilter())) {
+              Filter f = null;
+              for (Filter filter : query.getFilters()) {
```

Comment:
Using streams here should simplify the code and remove the null check.

Reply:
yeah I may switch to that since this isn't a critical path, just testing.